app/models/programme_activity_grouping.rb

```diff
@@ -69,4 +70,12 @@ def order_programme_activities_for_user(user, pathway)
 
     completed_legacy_activities + completed_non_legacy_activities + non_completed_non_legacy_activities
   end
+
+  def objective_displayed_in?(role)
+    role.in? %i[progress_bar body]
+  end
+
+  def progress_bar_path
```

Comment:
The progress_bar_path and its corresponding target in the DOM is a private contract of this object.

If it's rendered in the progress_bar, it should also be rendered in the body which will have the matching ID on an element